base/set.jl

```diff
@@ -502,16 +512,27 @@ function allunique(t::Tuple)
 end
 allunique(t::Tuple{}) = true
 
+function allunique(f::F, t::Tuple) where {F}
+    length(t) < 2 && return true
+    length(t) < 32 || return _hashed_allunique(Generator(f, t))
```

Comment:
Is `_hashed_allunique` intentional here?

Reply:
Can you say why you think it might not be? I feel like I'm missing something.

Reply:
No, you are not missing anything. I thought `_hashed_allunique` gets invoked for short tuples (confused `||` for `&&`). Sorry for the noise!